orbit-2-core/README.md

```diff
@@ -0,0 +1,319 @@
+# Orbit 2 Core
+
+This is the core module for the Orbit 2 framework. It's worth familiarising
+yourself with the [overview documentation](docs/orbit-2-overview.md) before
+reading this section.
+
+The core module provides all the basic parts of Orbit:
+
+- Public API
+- Core DSL
+- Plugin registry
+- Container implementation
+
+## Architecture
+
+![Orbit 2 overview 1](docs/orbit-2-overview-1.svg)
+
+This diagram shows a simple representation of how an Orbit system (or similar
+systems like MVI/Redux/Cycle) works in simple principles.
+
+1. The UI sends actions asynchronously to some business component
+1. The business component applies some business logic to the incoming events
+1. The business component then emits some events further down the chain
+1. Every event is reduced with the current state of the system to produce a new
+   state
+1. The state is then emitted back to the UI which renders itself based upon
+   information within
+
+The main thing to remember is that the UI cannot make any business decisions
+by itself. It should know only how to render itself based on the input state.
+
+### Orbit concepts
+
+![Orbit 2 overview 2](docs/orbit-2-overview-2.svg)
+
+We can map the above logic onto real components.
+
+1. UI invokes functions on a class implementing the `ContainerHost` interface
+1. The functions call through to a `Container` instance through the `orbit`
+   DSL function
+1. Transformers apply business logic that transforms function parameters into
+   single or multiple events
+1. The reducer reduces the current state of the system with the incoming events
+   to produce new states
+1. The new state is sent to observers
+
+Notes:
+
+- UI can be an `Activity`, `Fragment` or a simple `View`. An Orbit system can
```

Comment:
I'd move this note into the statement about UI above.

```
1. UI invokes functions on a class implementing the `ContainerHost` interface. Typically in Android this might be an Activity, Fragment or a simple View. However, an Orbit system can also be run without any UI, for example as a background service.
```